ui/common/Hasher.tsx

```diff
@@ -0,0 +1,154 @@
+'use client'
+
+import clsx from 'clsx'
+import { useEffect, useState } from 'react'
+import { Button } from 'shared'
+import sjcl from 'sjcl'
+
+export default function Hasher({
+  answer,
+  label,
+  onChange,
+  auto,
+  input,
+}: {
+  answer: string | number
+  label: string
+  onChange: any
+  auto: boolean
+  input: string
+}) {
+  const [hash, setHash] = useState('')
+  const [clicked, setClicked] = useState(false)
+
+  const handleAutoChange = () => {
+    let randomNum = (Math.random() * 1000000000000000000).toString()
+    setHash(sjcl.codec.hex.fromBits(sjcl.hash.sha256.hash(randomNum)))
+    console.log(randomNum)
+  }
+
+  const handleHash = (input) => {
+    if (input) {
+      const bitArray = sjcl.hash.sha256.hash(input)
+      setHash(sjcl.codec.hex.fromBits(bitArray))
+    } else {
+      setHash('')
+    }
+  }
+
+  function handleClick() {
+    setClicked(!clicked)
+  }
+
+  useEffect(() => {
+    const miner =
+      clicked &&
+      typeof answer === 'string' &&
+      !hash.startsWith(answer) &&
+      setInterval(() => {
+        handleAutoChange()
+      }, 10)
+    !clicked && handleHash(input)
+    if (typeof answer === 'string') {
+      onChange(hash.substring(0, answer.length))
+    } else {
+      onChange(hash)
+    }
+    return () => {
+      clearInterval(miner)
+    }
+  })
```

Comment:
No dependency array means this runs on every render, seems odd considering you have an interval (though you do clear it propery), can you explain what this does?